### PR TITLE
Provide a service to enable network interfaces on boot.

### PR DESCRIPTION
**Pull request by illfelder**
- Update the google-compute-engine-init package to run the service.
- Create a common library for inspecting network interfaces.
- Improved style consistency.
- Fixed the instance config file logic.

### CODE REVIEW

File: google_compute_engine/network_setup/network_setup.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# limitations under the License.
+
+"""Manage IP forwarding on a Google Compute Engine instance.
+
+Fetch a list of public endpoint IPs from the metadata server, compare it with
+the IPs configured on eth0, and add or remove addresses from eth0 to make them
+match. Only remove those which match our proto code.
+
+Command used to add IPs:
+  ip route add to local $IP/32 dev eth0 proto 66
```
Comment (jon) [inline google_compute_engine/network_setup/network_setup.py]: Stale comment?

**Comment by zmarano** [comment]
LGTM

**Comment by illfelder** [comment]
Updated and tested. Please take a look before I merge.

**Comment by zmarano** [comment]
LGTM

**Pull request merged**